lib/shopify_api/session.rb

```diff
@@ -73,9 +73,10 @@ def initialize(url, token = nil, extra = {})
       self.extra = extra
     end
 
-    def create_permission_url(scope, redirect_uri = nil)
-      params = {:client_id => api_key, :scope => scope.join(',')}
+    def create_permission_url(scope, redirect_uri = nil, state: nil)
```

Comment:
Moving to named parameters seemed better long term but makes for sort of a wonky interface right now.

Reply:
What would we think about instead consolidating into an `options` hash? CC @jamiemtdwyer 

```rb
def create_permission_url(scope, options = {})
# ...
end

create_permission_url(["read_orders"], state: "foobar", redirect_uri: "baz.myshopify.com")
```

Reply:
Consolidating into an options hash isn't a bad idea. I like it because it's future proof, it allows extending this in the future to support generating online access tokens without changing the public API again. That being said, I think I would keep `redirect_uri` and actually remove the defaulting to `nil`, since redirect_uri is required now.

I don't much like the idea of mixing keyword and positional arguments.

Reply:
@jamiemtdwyer Just so I'm clear before I try to go get all those tests changed around: You're okay with a breaking to the interface that makes the `redirect_uri` a required parameter? So we'll have a method like:
```rb
    def create_permission_url(scope, redirect_uri, state: nil)
      params = { client_id: api_key, scope: scope.join(','), redirect_uri: redirect_uri }
      params[:state] = state if state
      "#{site}/oauth/authorize?#{parameterize(params)}"
    end
```

Reply:
Do I need to make changes to the readme or were you planning on handling that?

Reply:
Changes to the README where necessary would be super appreciated as part of this PR